### DISCUSSION TITLE
Alpine $persist returns [object] instead of storage value in Alpine.store

### DISCUSSION THREAD

**Discussion by arthurshlain** [alpinejs/alpine#4248] [1. Help] [answered]
I'm trying to use the [Alpine.$persist global](https://alpinejs.dev/plugins/persist#using-alpine-persist-global), but got object instead of string:

![image](https://github.com/alpinejs/alpine/assets/2047645/154ab478-f4cd-44e6-a456-f8c0e3a74459)

The value with specified key `accessibility_fontSizeClassName` is not present in Local storage.

accessibility.js
```js
import Alpine from "alpinejs"

import persist from "@alpinejs/persist"

Alpine.plugin(persist)

Alpine.store('accessibility', {

    activeFontSizeClassName: Alpine.$persist('text-[16px]').as('accessibility_fontSizeClassName'),

    init() {

        console.log(this.activeFontSizeClassName);
        // Proxy(Object) {initialValue: 'text-[16px]', _x_interceptor: true, …}

        document.documentElement.classList.add(this.activeFontSizeClassName);
        // ↑ (string expected)
        // Uncaught DOMException: Failed to execute 'add' on 'DOMTokenList': 
        // The token provided ('[object Object]') contains HTML space characters, 
        // which are not valid in tokens.
    },

});
```

app.js
```js
import Alpine from 'alpinejs'

// … import other Alpine plugins

import './store/accessibility'

// … import data objects

window.Alpine = Alpine

Alpine.start()
```

    "@alpinejs/anchor": "^3.14.0",
    "@alpinejs/collapse": "^3.14.0",
    "@alpinejs/focus": "^3.14.0",
    "@alpinejs/intersect": "^3.14.0",
    "@alpinejs/persist": "^3.14.0",
    "@tailwindcss/forms": "^0.5.7",
    "alpinejs": "^3.14.0",
    "axios": "^1.6.8",
    "cross-env": "^7.0.3",
    "fast-glob": "^3.3.2",
    "js-cookie": "^3.0.5",
    "laravel-mix": "^6.0.49",
    "swiper": "^11.1.4",
    "tailwindcss": "^3.4.1"

**Comment by SimoTod** [comment, marked as answer]
`init` runs too early when persist is not the persisted value yet. You need to use nextTick
```javascript
init() {
  Alpine.nextTick(() => {
    console.log(this.activeFontSizeClassName); 
    document.documentElement.classList.add(this.activeFontSizeClassName);
  })
}
```

**Reply by arthurshlain** [reply]
Thank you!

**Reply by ekwoka** [reply]
feels like interceptors should be handled before init is called...

**Reply by SimoTod** [reply]
> feels like interceptors should be handled before init is called...

They do for x-data, they don't for stores. Probably an oversight, it should be just a matter of moving one line further up, to be fair

**Reply by bashunaimiroy** [reply]
Hi @SimoTod.  Hope you're well. 

I'm experiencing the same issue posted by OP. Could you explain what you mean by the 'moving one line further up'? Which line? And is this is a proposed fix in our code, or something that you're suggesting could be fixed in a PR to Alpine source? 

Thanks

**Reply by SimoTod** [reply]
That comment was about how to address the bug in the Alpine source code. The fix in userland is to use nextTick

**Reply by ekwoka** [reply]
I'll just make that PR today 🥱

**Comment by ekwoka** [comment]
#4278
